modules/core/src/test/java/org/apache/synapse/config/xml/MultiXMLConfigurationSerializerTest.java

```diff
@@ -0,0 +1,245 @@
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+package org.apache.synapse.config.xml;
+
+import org.apache.axiom.om.OMNode;
+import org.apache.synapse.config.SynapseConfiguration;
+import org.apache.synapse.core.axis2.ProxyService;
+import org.apache.synapse.mediators.TestUtils;
+import org.apache.synapse.mediators.base.SequenceMediator;
+import org.apache.synapse.mediators.template.TemplateMediator;
+import org.apache.synapse.registry.Registry;
+import org.apache.synapse.registry.SimpleInMemoryRegistry;
+import org.apache.synapse.startup.quartz.QuartzTaskManager;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This is the test class for MultiXMLConfigurationSerializer class.
+ */
+public class MultiXMLConfigurationSerializerTest {
+
+    private static final String SYNAPSE_XML = "synapse.xml";
+    private static final String KEY_DYNAMIC_SEQUENCE_1 = "dynamic_sequence_1";
+    private static final String KEY_DYNAMIC_ENDPOINT_1 = "dynamic_endpoint_1";
+    private static final String TEST_DIRECTORY_NAME = "serializeTestFolder";
+
+    private static final String DYNAMIC_ENDPOINT_1 =
+            "<endpoint xmlns=\"http://ws.apache.org/ns/synapse\">\n" +
+                    "    <address uri=\"http://test.url\"/>\n" +
+                    "</endpoint>";
+
+    private static final String DYNAMIC_SEQUENCE_1 =
+            "<sequence xmlns=\"http://ws.apache.org/ns/synapse\" name=\"seq1\">\n" +
+                    "    <property name=\"foo\" value=\"bar\" />" +
+                    "</sequence>";
+
+    /**
+     * Test serialize method and assert synapse.xml is created.
+     */
+    @Test
+    public void testSerialize() {
+        MultiXMLConfigurationSerializer serializer = new MultiXMLConfigurationSerializer(TEST_DIRECTORY_NAME);
+        SynapseConfiguration configuration = new SynapseConfiguration();
+        serializer.serialize(configuration);
+        Assert.assertTrue("Error in serializing Synapse configuration.", new File(TEST_DIRECTORY_NAME +
+                File.separator + SYNAPSE_XML).exists());
+        removeTestFolder(TEST_DIRECTORY_NAME);
+    }
+
+    /**
+     * Test serialize method with root directory already existing and assert synapse.xml is created.
+     *
+     * @throws IOException - IOException in file creation.
+     */
+    @Test
+    public void testSerialize2() throws IOException {
+        File file = new File(TEST_DIRECTORY_NAME);
+        if (!file.exists()) {
+            file.mkdir();
+            File xmlFile = new File(TEST_DIRECTORY_NAME + File.separator + "test.xml");
+            if (!xmlFile.exists()) {
+                xmlFile.createNewFile();
+            }
+        }
+        MultiXMLConfigurationSerializer serializer = new MultiXMLConfigurationSerializer(TEST_DIRECTORY_NAME);
```

Comment:
xmlFile.createNewFile() returns before the file is actually created in the file system. So it's better to check if the file exsists before proceeding further.

Reply:
fixed this